csharp/src/Ice/Object.cs

```diff
@@ -27,40 +29,40 @@ public interface Object : ICloneable
     /// </summary>
     ///
     /// <param name="s">The type ID of the Slice interface to test against.</param>
-    /// <param name="current">The Current object for the invocation.</param>
+    /// <param name="current">The Current object for the dispatch.</param>
     /// <returns>True if this object has the interface
     /// specified by s or derives from the interface specified by s.</returns>
-    bool ice_isA(string s, Current current = null);
+    bool ice_isA(string s, Current current);
 
     /// <summary>
     /// Tests whether this object can be reached.
     /// </summary>
-    /// <param name="current">The Current object for the invocation.</param>
-    void ice_ping(Current current = null);
+    /// <param name="current">The Current object for the dispatch.</param>
+    void ice_ping(Current current);
 
     /// <summary>
     /// Returns the Slice type IDs of the interfaces supported by this object.
     /// </summary>
-    /// <param name="current">The Current object for the invocation.</param>
+    /// <param name="current">The Current object for the dispatch.</param>
     /// <returns>The Slice type IDs of the interfaces supported by this object, in alphabetical order.</returns>
-    string[] ice_ids(Current current = null);
+    string[] ice_ids(Current current);
 
     /// <summary>
     /// Returns the Slice type ID of the most-derived interface supported by this object.
     /// </summary>
-    /// <param name="current">The Current object for the invocation.</param>
+    /// <param name="current">The Current object for the dispatch.</param>
     /// <returns>The Slice type ID of the most-derived interface.</returns>
-    string ice_id(Current current = null);
+    string ice_id(Current current);
 
     /// <summary>
-    /// Dispatches an invocation to a servant. This method is used by dispatch interceptors to forward an invocation
+    /// Dispatches a dispatch to a servant. This method is used by dispatch interceptors to forward a dispatch
```

Comment:
I don't really like "Dispatches a dispatch", there's no type to match this and it sounds odd.

The type is a request, I think it makes sense to just say that.
```suggestion
    /// Dispatches a request to a servant. This method is used by dispatch interceptors to forward a dispatch
```